Zotero/Scenes/Detail/Trash/ViewModels/TrashActionHandler.swift

```diff
@@ -0,0 +1,663 @@
+        coordinator: DbCoordinator,
+        viewModel: ViewModel<TrashActionHandler>
+    ) throws -> TrashState.Snapshot {
+        let searchComponents = searchTerm.flatMap({ createComponents(from: $0) }) ?? []
+        let itemsRequest = ReadItemsDbRequest(collectionId: .custom(.trash), libraryId: libraryId, filters: filters, sortType: sortType, searchTextComponents: searchComponents)
+        let items = try coordinator.perform(request: itemsRequest)
+        var collections: Results<RCollection>?
+        if filters.isEmpty {
+            let collectionsRequest = ReadCollectionsDbRequest(libraryId: libraryId, trash: true, searchTextComponents: searchComponents)
+            collections = (try coordinator.perform(request: collectionsRequest)).sorted(by: collectionSortDescriptor(for: sortType))
+        }
+        let (keys, keyToIdx) = createSnapshotData(fromItems: items, collections: collections, sortType: sortType)
+        let (itemsToken, collectionsToken) = observe(items: items, collections: collections, viewModel: viewModel)
+        return TrashState.Snapshot(sortedKeys: keys, keyToIdx: keyToIdx, itemResults: items, itemsToken: itemsToken, collectionResults: collections, collectionsToken: collectionsToken)
+
+        func collectionSortDescriptor(for sortType: ItemsSortType) -> [RealmSwift.SortDescriptor] {
+            switch sortType.field {
+            case .dateModified:
+                return [
+                    SortDescriptor(keyPath: "dateModified", ascending: sortType.ascending),
+                    SortDescriptor(keyPath: "sortName", ascending: sortType.ascending)
+                ]
+
+            case .title, .creator, .date, .dateAdded, .itemType, .publisher, .publicationTitle, .year:
+                return [SortDescriptor(keyPath: "sortName", ascending: sortType.ascending)]
+            }
+        }
+    }
+
+    private func createSnapshotData(fromItems items: Results<RItem>?, collections: Results<RCollection>?, sortType: ItemsSortType) -> ([TrashKey], [TrashKey: Int]) {
+        var itemsIdx = 0
+        var collectionsIdx = 0
+        var keys: [TrashKey] = []
+        var keyToIdx: [TrashKey: Int] = [:]
+        if let collections, let items {
+            while itemsIdx < items.count && collectionsIdx < collections.count {
+                let item = items[itemsIdx]
+                let collection = collections[collectionsIdx]
+                if areInIncreasingOrder(lObject: item, rObject: collection, sortType: sortType) {
+                    keys.append(TrashKey(type: .item, key: item.key))
+                    keyToIdx[keys.last!] = itemsIdx
+                    itemsIdx += 1
+                } else {
+                    keys.append(TrashKey(type: .collection, key: collection.key))
+                    keyToIdx[keys.last!] = collectionsIdx
+                    collectionsIdx += 1
+                }
+            }
+        }
+        if let collections, collectionsIdx < collections.count {
```

Comment:
Can be simplified to:
```suggestion
        if let collections {
```